_meeting-notes/2019-10-22.md

```diff
@@ -0,0 +1,128 @@
+Test Ground Weekly
+===
+
+###### tags: `IPFS` `Testing`
+
+:::info
+- **Host:** @daviddias
+- **Location:** https://protocol.zoom.us/my/daviddias
+- **Date:** Oct 22, 2019 5:00 PM (UTC)
+- **Participants:**
+    - @daviddias
+    - @jimpick
+    - @stebalien
+    - @momack2
+- **Agenda**
+  1. Sprint Update `20min`
+  2. Identify blockers / Answer Questions `20min`
+  3. This week's planning `20min`
+- **Reference:** 
+  - [Last week meeting minute](https://github.com/ipfs/testground/pull/69)
+  - [Trello Board](https://trello.com/b/O9z3ljaH/testground-ignition-%F0%9F%9A%80)
+  - [Testing Infra Design](https://github.com/ipfs/testground/blob/master/docs/SPEC.md)
+  - [Testing Inventory](https://github.com/ipfs/testground/blob/master/docs/test-inventory.md)
+  - [Testing Improvement Tracker](https://docs.google.com/spreadsheets/d/1xyqyGUF-oe3x9ln88YonVeOMWWdknik74lVgL_3dBY8/edit#gid=0)
+
+:::
+
+## :mega: Update (Done, Shipped, Made Progress)
+
+- ==@daviddias==
+  - Completed the "go-ipfs as a Library" Tutorial, now in review phase by the docs Team https://github.com/ipfs/go-ipfs/pull/6695/
+  - Landed smlbench working with TestGround orchestrator (thank you Jim for helping with the last mile there ❤️) https://trello.com/c/1LTXALLW/62-get-smlbench-plan-to-work-with-testground-orchestrator
+  - Continue iterating on the Test Plan spec  https://trello.com/c/0erPZId8/43-specify-10-test-plans
+    - now with differentiation for Plan & Test Parameters https://github.com/ipfs/testground/pull/76
+    - now each Test Plan lives with the plan folder of the Plan itself https://github.com/ipfs/testground/pull/77
+  - Revisited the OKRs as requested https://trello.com/c/o09gs18G/63-review-and-iterate-on-q4-okrs
+  - Created a draft Roadmap for review https://github.com/ipfs/testground/pulls
+  - Got sick, then got better :) ETOOMUCHLONGFLIGHTS
+  - Broke a tooth 🦷 (tip: watch out for those Swiss Ricolas!). **FYI** will be traveling this week to Lisbon for a emergency dentist appt (and a reschedule of others I was going to reschedule to post hack week, but this way I merge it all together rather than going to Lisbon 2x). This means that I'm almost at 2 days of lost work given flying + transportation + being at the dentist.
+- ==@jimpick==
+  - smlbench (adds a randomly generated file on one node, gets it from a 2nd node)
+    - initial work, one docker container, IPTB with 2 IPFS nodes https://github.com/ipfs/testground/pull/74
+    - follow-on work, two dockers containers, IPTB with 1 IPFS node per container, using Sync SDK (redis) to co-ordinate https://github.com/ipfs/testground/pull/74
+  - Lots of little puzzles solved:
+    - IPFS into Docker using Docker.template and conditional via shell script (works, but not satisfied)
+    - IPTB SDK usage
```

Comment:
What's this @jimpick ?

Reply:
The test plan uses IPTB via Testground SDK documented here:

https://godoc.org/github.com/ipfs/testground/sdk/iptb

Rather than spin up a whole bunch of nodes, we're only using IPTB to spin up a single node (per Docker container) that we can control using the go-ipfs-api documented here:

https://godoc.org/github.com/ipfs/go-ipfs-api

Since all that infrastructure already existed as part of IPTB, it made sense to re-use it.

I think the "go-ipfs as a library" approach is also of high value - where we build IPFS directly into the test instead of calling an existing binary - that's probably useful for when we need deeper control or need to test a specific subset of things. https://github.com/ipfs/go-ipfs/pull/6695/